thoth/metrics_exporter/jobs.py

```diff
@@ -64,55 +69,89 @@ def countJobStatus(JobListItems: dict) -> (int, int, int):
-        "https://paas.upshift.redhat.com:443/apis/batch/v1",
-        namespace
+        "https://paas.upshift.redhat.com:443/apis/batch/v1", namespace
     )  # FIXME the OpenShift API URL should not be hardcoded
 
     try:
         # FIXME we should not hardcode the solver dist names
         response = requests.get(
             endpoint,
             headers={
-                'Authorization': 'Bearer {}'.format(get_service_account_token()),
-                'Content-Type': 'application/json'
+                "Authorization": "Bearer {}".format(get_service_account_token()),
+                "Content-Type": "application/json",
             },
-            params={'labelSelector': 'component=solver-f27'},
-            verify=False
+            params={"labelSelector": "component=solver-f27"},
+            verify=False,
         ).json()
 
-        created, failed, succeeded = countJobStatus(response['items'])
+        created, failed, succeeded = countJobStatus(response["items"])
+
+        solver_jobs_total.labels("f27", "created").set(created)
+        solver_jobs_total.labels("f27", "failed").set(failed)
+        solver_jobs_total.labels("f27", "succeeded").set(succeeded)
+
+    except ResourceNotFoundError as excptn:
+        _LOGGER.error(excptn)
+
+
+@solver_documents_seconds.time()
+def get_solver_documents(solver_name: str = None):
+    """Get the total number Solver Documents in Graph Database."""
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    try:
+        graph_db = GraphDatabase.create("janusgraph", port=8182)
+        graph_db.connect()
+    except aiohttp.client_exceptions.ClientConnectorError as excptn:
+        _LOGGER.error(excptn)
+        return
+
+    solver_documents = graph_db.get_solver_documents_count()
+
+    _LOGGER.debug("solver_documents=%r", solver_documents)
+
+
+@analyzer_documents_seconds.time()
+def get_analyzer_documents():
+    """Get the total number Analyzer Documents in Graph Database."""
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    graph_db = GraphDatabase.create("janusgraph", port=8182)
+    graph_db.connect()
+    analyzer_documents = graph_db.get_solver_documents_count()
 
-        thoth_solver_jobs_total.labels('f27', 'created').set(created)
-        thoth_solver_jobs_total.labels('f27', 'failed').set(failed)
-        thoth_solver_jobs_total.labels('f27', 'succeeded').set(succeeded)
+    _LOGGER.debug("analyzer_documents_total=%r", analyzer_documents)
 
-    except ResourceNotFoundError as excpt:
-        _LOGGER.error(excpt)
 
 def get_janusgraph_v_and_e_total():
     """Get the total number of Vertices and Edges stored in JanusGraph Server."""
-    graph_db = GraphDatabase.create('test.janusgraph.thoth-station.ninja', port=8182)
-    graph_db.connect() # FIXME no hardcoded hostnames
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    graph_db = GraphDatabase.create("janusgraph", port=8182)
```

Comment:
a TODO for not hardcoded name?

Reply:
well it's the service name and service port, this is provided/injected by OpenShift and is always present... all other services depend on it too

Reply:
So pick it from env variables (that's what I was referring to)?

Reply:
ah, right... you refer to https://github.com/thoth-station/storages/blob/master/thoth/storages/graph/janusgraph.py#L94